File: scripts/check_nav_v2_manager_confirmed_results.py
#!/usr/bin/env python3
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]

manager = (ROOT / "assets/js/nav-v2/manager-v2.js").read_text(encoding="utf-8")
model = (ROOT / "assets/js/nav-v2/manager-confirmed-results-model-v2.js").read_text(encoding="utf-8")
html = (ROOT / "manager-v2.html").read_text(encoding="utf-8")
css = (ROOT / "assets/css/nav-v2-manager.css").read_text(encoding="utf-8")
semantic = (ROOT / "scripts/check-nav-v2-manager-confirmed-results.mjs").read_text(encoding="utf-8")
workflow = (ROOT / ".github/workflows/nav-v2-manager-confirmed-results.yml").read_text(encoding="utf-8")

required_manager_markers = [
    "manager-confirmed-results-model-v2.js?v=20260715-01",
    "managerResultCandidate(item, { now, maxAgeDays: 7 })",
    "rpc('nav_v2_get_deal_card', { p_deal_id: item.deal_id }, 20000)",
    "COMPLETION_CONCURRENCY = 4",
    "COMPLETION_LOAD_LIMIT = 40",
    "Подтверждённые результаты",
    "data-confirmed-filter",
    "Серверное подтверждение",
    "Следующий ответственный шаг",
    "Открыть следующий шаг",
]
for marker in required_manager_markers:
    assert marker in manager, f"manager-v2.js marker missing: {marker}"

assert "nav_v2_update_" not in manager, "Manager result control must remain read-only"
assert manager.count("nav_v2_get_operational_readiness_preview") == 1
assert manager.count("nav_v2_get_deal_card") == 1
assert ".filter((item) => managerResultCandidate" in manager
assert ".slice(0, COMPLETION_LOAD_LIMIT)" in manager

required_model_markers = [
    "buildDealCompletionEvidence",
    "DEFAULT_MAX_AGE_DAYS = 7",
    "DEFAULT_TIME_ZONE = 'Europe/Moscow'",
    "window: completedDay && completedDay === today ? 'today' : 'recent'",
    "nextHref",
    "actorKnown",
    "serverEventType",
    "summarizeManagerConfirmedResults",
]
for marker in required_model_markers:
    assert marker in model, f"confirmed results model marker missing: {marker}"

assert "nav-v2-manager.css?v=20260715-01" in html
assert "manager-v2.js?v=20260715-01" in html
for marker in [
    ".manager-confirmed-results",
    ".manager-confirmed-card",
    ".manager-confirmed-meta",
    ".manager-confirmed-next",
    ".manager-confirmed-actions .btn",
]:
    assert marker in css, f"manager CSS marker missing: {marker}"

for marker in [
    "Previous Moscow calendar day must not be shown as today",
    "Assignment-only/no-op event must not masquerade as completed work",
    "sevenDays: 2",
]:
    assert marker in semantic, f"semantic regression marker missing: {marker}"

for marker in [
    "python3 scripts/check_nav_v2_manager_confirmed_results.py",
    "node scripts/check-nav-v2-manager-confirmed-results.mjs",
    "node scripts/check-nav-v2-manager-action-routes.mjs",
    "python3 scripts/check_nav_v2_operational_readiness.py",
    "node --check assets/js/nav-v2/manager-confirmed-results-model-v2.js",
]:
    assert marker in workflow, f"workflow marker missing: {marker}"

print("Navigator v2 manager confirmed results static contract passed")

File: scripts/check-nav-v2-manager-confirmed-results.mjs
import assert from 'node:assert/strict';
import {
  buildManagerConfirmedResult,
  managerResultCandidate,
  sortManagerConfirmedResults,
  summarizeManagerConfirmedResults
} from '../assets/js/nav-v2/manager-confirmed-results-model-v2.js';

const now = '2026-07-15T12:00:00Z';
const profile = { id: 'owner-1', role: 'owner', full_name: 'Алексей Ковтун' };
const item = {
  deal_id: 'deal-1',
  title: 'Квартира — Чкалова, 4',
  card_url: './deal-card-v2.html?id=deal-1',
  last_activity_at: '2026-07-15T10:00:00Z'
};
const deal = {
  id: 'deal-1',
  status: 'preparing_deal',
  next_action: 'Назначить дату сделки',
  manager_id: 'manager-1',
  seller_spn_id: 'spn-1'
};

assert.equal(managerResultCandidate(item, { now, maxAgeDays: 7 }), true);
assert.equal(managerResultCandidate({ ...item, last_activity_at: '2026-07-01T10:00:00Z' }, { now, maxAgeDays: 7 }), false);

const todayResult = buildManagerConfirmedResult(item, {
  deal,
  tasks: [
    { id: 'task-done', title: 'Проверить пакет', status: 'done', completed_by: 'owner-1' },
    { id: 'task-next', title: 'Согласовать дату', status: 'open', priority: 'urgent', due_date: '2026-07-16', assigned_role: 'manager' }
  ],
  documents: [],
  risks: [],
  events: [{
    id: 'event-task',
    actor_id: 'owner-1',
    event_type: 'task_status_changed',
    created_at: '2026-07-15T10:00:00Z',
    event_data: { task_id: 'task-done', old_status: 'open', status: 'done' }
  }]
}, profile, { now, maxAgeDays: 7, timeZone: 'Europe/Moscow' });

assert.equal(todayResult.visible, true);
assert.equal(todayResult.window, 'today');
assert.equal(todayResult.kind, 'task');
assert.equal(todayResult.actor, 'Алексей Ковтун');
assert.match(todayResult.resultTitle, /Проверить пакет/);
assert.equal(todayResult.nextAction.title, 'Согласовать дату');
assert.equal(todayResult.nextAction.responsible, 'Менеджер');
assert.equal(todayResult.nextHref, './deal-card-v2.html?id=deal-1#tasks');
assert.match(todayResult.serverFact, /текущий статус задачи на сервере/);

const recentResult = buildManagerConfirmedResult({
  ...item,
  deal_id: 'deal-2',
  title: 'Дом — Прибрежная, 1',
  card_url: './deal-card-v2.html?id=deal-2',
  last_activity_at: '2026-07-14T18:00:00Z'
}, {
  deal: { ...deal, id: 'deal-2' },
  tasks: [],
  documents: [{ id: 'doc-1', title: 'Выписка ЕГРН', status: 'checked', checked_by: 'lawyer-1', responsible_role: 'lawyer' }],
  risks: [],
  events: [{
    id: 'event-doc',
    actor_id: 'lawyer-1',
    event_type: 'document_workflow_updated',
    created_at: '2026-07-14T18:00:00Z',
    event_data: { document_id: 'doc-1', old_status: 'received', status: 'checked' }
  }]
}, profile, { now, maxAgeDays: 7, timeZone: 'Europe/Moscow' });

assert.equal(recentResult.visible, true);
assert.equal(recentResult.window, 'recent', 'Previous Moscow calendar day must not be shown as today');
assert.equal(recentResult.kind, 'document');
assert.equal(recentResult.actor, 'Юрист');

const noOp = buildManagerConfirmedResult(item, {
  deal,
  tasks: [],
  documents: [{ id: 'doc-1', title: 'Выписка ЕГРН', status: 'checked' }],
  risks: [],
  events: [{
    id: 'event-doc-noop',
    actor_id: 'lawyer-1',
    event_type: 'document_workflow_updated',
    created_at: '2026-07-15T09:00:00Z',
    event_data: { document_id: 'doc-1', old_status: 'checked', status: 'checked' }
  }]
}, profile, { now, maxAgeDays: 7, timeZone: 'Europe/Moscow' });
assert.equal(noOp.visible, false, 'Assignment-only/no-op event must not masquerade as completed work');

const stale = buildManagerConfirmedResult(item, {
  deal,
  tasks: [{ id: 'task-done', title: 'Старая задача', status: 'done' }],
  documents: [],
  risks: [],
  events: [{
    event_type: 'task_status_changed',
    created_at: '2026-07-01T10:00:00Z',
    event_data: { task_id: 'task-done', old_status: 'open', status: 'done' }
  }]
}, profile, { now, maxAgeDays: 7, timeZone: 'Europe/Moscow' });
assert.equal(stale.visible, false);

const sorted = sortManagerConfirmedResults([recentResult, todayResult, noOp]);
assert.deepEqual(sorted.map((result) => result.dealId), ['deal-1', 'deal-2']);
assert.deepEqual(summarizeManagerConfirmedResults(sorted), { today: 1, sevenDays: 2 });

console.log('Navigator v2 manager confirmed results semantic checks passed');
